src/Dropdown/Dropdown.js

```diff
@@ -111,55 +116,67 @@ export default class Dropdown extends React.Component {
     );
   };
 
-  toggle = e => {
-    if (e) e.stopPropagation();
-
+  toggle = () => {
     if (this.state.isOpen) {
       this.hide();
     } else {
       this.show();
     }
   };
 
-  attachAutocloseListener() {
-    if (!this.props.autoclose) {
-      return;
-    }
-    document.addEventListener('click', this.autoclose);
```

Comment:
It seems like we can rely on React's supported synthetic events here and not need to use `addEventListener`, which is generally discouraged in the [docs](https://reactjs.org/docs/handling-events.html).

> When using React you should generally not need to call addEventListener to add listeners to a DOM element after it is created. Instead, just provide a listener when the element is initially rendered.